NetKet/GroundState/pyground_state.hpp

```diff
@@ -31,33 +31,77 @@ namespace netket {
+  py::class_<VariationalMonteCarlo>(vmc, "Vmc")
       .def(py::init<const AbstractOperator &, SamplerType &,
-                    AbstractOptimizer &, int, int, std::string, int, int,
-                    std::string, double, bool, bool, bool, int>(),
+                    AbstractOptimizer &, int, int, int, const std::string &,
+                    double, bool, bool, bool>(),
            py::keep_alive<1, 2>(), py::keep_alive<1, 3>(),
            py::keep_alive<1, 4>(), py::arg("hamiltonian"), py::arg("sampler"),
-           py::arg("optimizer"), py::arg("n_samples"), py::arg("niter_opt"),
-           py::arg("output_file"), py::arg("discarded_samples") = -1,
+           py::arg("optimizer"), py::arg("n_samples"),
+           py::arg("discarded_samples") = -1,
            py::arg("discarded_samples_on_init") = 0, py::arg("method") = "Sr",
            py::arg("diag_shift") = 0.01, py::arg("rescale_shift") = false,
-           py::arg("use_iterative") = false, py::arg("use_cholesky") = true,
-           py::arg("save_every") = 50)
+           py::arg("use_iterative") = false, py::arg("use_cholesky") = true)
+      .def_property_readonly("psi", &VariationalMonteCarlo::GetPsi)
       .def("add_observable", &VariationalMonteCarlo::AddObservable,
            py::keep_alive<1, 2>())
-      .def("run", &VariationalMonteCarlo::Run);
-
-  py::class_<ImaginaryTimeDriver>(subm, "ImaginaryTimeDriver")
-      .def(py::init<ImaginaryTimeDriver::Matrix &,
-                    ImaginaryTimeDriver::Stepper &, JsonOutputWriter &, double,
-                    double, double>(),
-           py::arg("hamiltonian"), py::arg("stepper"), py::arg("output_writer"),
-           py::arg("tmin"), py::arg("tmax"), py::arg("dt"))
-      .def("add_observable", &ImaginaryTimeDriver::AddObservable,
+      .def("run", &VariationalMonteCarlo::Run, py::arg("filename_prefix"),
+           py::arg("max_steps") = nonstd::nullopt, py::arg("step_size") = 1,
+           py::arg("save_params_every") = 50)
+      .def("iter", &VariationalMonteCarlo::Iterate,
+           py::arg("max_steps") = nonstd::nullopt, py::arg("step_size") = 1,
+           py::arg("store_params") = true);
+
+  py::class_<VariationalMonteCarlo::Step>(vmc, "Step")
+      .def_readonly("current_step", &VariationalMonteCarlo::Step::index)
+      .def_readonly("acceptance", &VariationalMonteCarlo::Step::acceptance)
+      .def_readonly("observables", &VariationalMonteCarlo::Step::observables)
+      .def_readonly("params", &VariationalMonteCarlo::Step::parameters)
+      .def("__repr__", [](const VariationalMonteCarlo::Step &self) {
+        std::stringstream str;
+        str << "<VmcStep: step=" << self.index << ">";
+        return str.str();
+      });
+
+  py::class_<VariationalMonteCarlo::Iterator>(vmc, "Iterator")
+      .def("__iter__", [](VariationalMonteCarlo::Iterator &self) {
+        return py::make_iterator(self.begin(), self.end());
+      });
+
+  auto excact = subm.def_submodule("exact");
```

Comment:
same here, we could do `netket.ed.ImagTimePropagation` and `netket.ed.SparseDiag` etc?